### PR TITLE
Validate ServiceProvider in tests, updates to Spring Boot configuration provider

### PR DESCRIPTION
**Pull request by bart-vmware**
## Description

What started as turning on service provider validation in tests, identified several issues this PR fixes.

See the commit messages for details. For Steeltoe users, the following changes are noteworthy:
- `AddSteeltoe` now auto-wires the Spring Boot configuration providers and the dynamic console logger, if the related assemblies are loaded.
- Spring boot command-line configuration provider has been optimized. It no longer wraps the full configuration, but only affects command-line parameters. As a result, the extension method takes `string[] args` instead of `IConfiguration`. It is usually sufficient to just pass `args` that's implicitly in scope of `Program.cs`, but when called from elsewhere, use: `System.Environment.GetCommandLineArgs().Skip(1).ToArray()` (skipping the first entry because that's the executable).
- Spring environment variables configuration provider now properly handles array keys, such as: `spring.some[0].key`.

## Quality checklist

- [x] Your code complies with our [Coding Style](https://github.com/SteeltoeOSS/.github/blob/main/contributing-docs/contributing-code-style.md).
- [x] You've updated unit and/or integration tests for your change, where applicable.
- [ ] You've updated documentation for your change, where applicable.
      If your change affects other repositories, such as [Documentation](https://github.com/SteeltoeOSS/Documentation), [Samples](https://github.com/SteeltoeOSS/Samples) and/or [MainSite](https://github.com/SteeltoeOSS/MainSite), add linked PRs here.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.
- [ ] You've added required license files and/or file headers (explaining where the code came from with proper attribution), where code is copied from StackOverflow, a blog, or OSS.

### CODE REVIEW

File: src/Management/test/Endpoint.Test/ManagementWebHostBuilderExtensionsTest.cs [hunk 1]
```diff
@@ -206,7 +194,7 @@ public async Task AddHeapDumpActuator_IWebHostBuilder_IStartupFilterFires()
     {
         if (Platform.IsWindows)
```
Comment (TimHess) [inline src/Management/test/Endpoint.Test/ManagementWebHostBuilderExtensionsTest.cs:195]: Does this need to still be checking for `Platform.IsWindows`? I think at a minimum it works on Linux too

**Review by TimHess — APPROVED** [review]
LGTM, though title/description should mention changes to SB command line config provider

File: src/Management/test/Endpoint.Test/ManagementWebHostBuilderExtensionsTest.cs (hunk 1 above)
Comment (bart-vmware) [reply src/Management/test/Endpoint.Test/ManagementWebHostBuilderExtensionsTest.cs:195]: I don't know, just left it as-is. Can you open an issue to investigate, if needed?

**Comment by bart-vmware** [comment]
> LGTM, though title/description should mention changes to SB command line config provider

I've updated the description. Feel free to merge this PR, so you can base your work on it.

File: src/Management/test/Endpoint.Test/ManagementWebHostBuilderExtensionsTest.cs (hunk 1 above)
Comment (TimHess) [reply src/Management/test/Endpoint.Test/ManagementWebHostBuilderExtensionsTest.cs:195]: opened #1361

**Pull request merged**